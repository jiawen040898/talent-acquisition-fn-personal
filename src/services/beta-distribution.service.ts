export const EXPONENTIAL_SMOOTHING_WEIGHT = 0.025; // Gamma

const smooth = (alpha: number, beta: number, gamma: number): number => {
    return alpha + gamma * (beta - alpha);
};

const getGammaForUniformSmoothing = (size: number): number => {
    return 1 / size;
};

const getNewMean = (
    oldMean: number,
    value: number,
    weight = EXPONENTIAL_SMOOTHING_WEIGHT,
): number => {
    return smooth(oldMean, value, weight);
};

const getNewVariance = (
    oldVariance: number,
    oldMean: number,
    newMean: number,
    value: number,
    weight = EXPONENTIAL_SMOOTHING_WEIGHT,
): number => {
    return smooth(oldVariance, (value - oldMean) * (value - newMean), weight);
};

const getAlphaAndBeta = (
    mean: number,
    variance: number,
    size: number,
): {
    alpha: number;
    beta: number;
} => {
    if (size === 1) {
        return {
            alpha: 1,
            beta: 1,
        };
    }

    const alpha = mean * ((mean * (1 - mean)) / variance - 1);
    const beta = (1 - mean) * ((mean * (1 - mean)) / variance - 1);
    return { alpha, beta };
};

export const BetaDistributionService = {
    getGammaForUniformSmoothing,
    getNewMean,
    getNewVariance,
    getAlphaAndBeta,
};
