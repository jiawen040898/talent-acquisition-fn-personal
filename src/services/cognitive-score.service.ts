import { JobApplicationScoreType } from '@pulsifi/constants';
import { JobApplicationQuestionnaireDto } from '@pulsifi/dtos';
import { DomainTraitOutcome, logger } from '@pulsifi/fn';
import {
    AssessmentRawResult,
    CognitiveQuestionnaireFramework,
} from '@pulsifi/interfaces';
import { JobApplicationScoreMapper } from '@pulsifi/mappers';
import { JobApplicationScore } from '@pulsifi/models';
import { JobApplicationScoreUtil, QuestionnaireUtil } from '@pulsifi/shared';
import { sum } from 'lodash';
import { In } from 'typeorm';

import { getDataSource } from '../database';
import { JobApplicationScoreService } from './job-application-score.service';

const processCognitiveScore = (
    assessment: JobApplicationQuestionnaireDto,
    jobApplicationId: string,
    updatedBy: number,
    recipeId?: string,
): JobApplicationScore => {
    const scoreType =
        JobApplicationScoreMapper.mapCognitiveQuestionnaireFrameworkToScoreType[
            assessment.questionnaire_framework as CognitiveQuestionnaireFramework
        ];

    const alias =
        JobApplicationScoreMapper.mapCognitiveScoreTypeToAlias[scoreType];

    const { domain_score: domainScore, domain_alias: domainAlias } = (
        assessment?.result_raw as AssessmentRawResult
    ).scores?.find(
        (result) => result.domain_alias === alias,
    ) as DomainTraitOutcome;

    return JobApplicationScoreMapper.mapJobApplicationScore(
        JobApplicationScoreUtil.getFormattedJobApplicationScore(
            domainScore,
            scoreType,
        ),
        {
            cognitive_result: {
                domain_score: domainScore || 0,
                domain_alias: domainAlias,
                domain_percentile: null,
                ingredient_weightage: null,
            },
        },
        scoreType,
        JobApplicationScoreService.getScoreDimension(domainScore),
        jobApplicationId,
        updatedBy,
        recipeId,
    );
};

const processReasoningAverage = async (
    newScores: JobApplicationScore[],
    jobApplicationId: string,
    updatedBy: number,
    recipeId?: string,
): Promise<JobApplicationScore> => {
    const dataSource = await getDataSource();
    const existingScores = await dataSource
        .getRepository(JobApplicationScore)
        .find({
            where: {
                job_application_id: jobApplicationId,
                score_type: In([
                    JobApplicationScoreType.REASONING_LOGICAL,
                    JobApplicationScoreType.REASONING_NUMERIC,
                    JobApplicationScoreType.REASONING_VERBAL,
                ]),
            },
        });

    const newScoreTypes = newScores.map((score) => score.score_type);

    const filteredExistingScores = existingScores.filter(
        (score) => !newScoreTypes.includes(score.score_type),
    );

    const availableCognitiveScores = [...newScores, ...filteredExistingScores];

    const scores = availableCognitiveScores.map((s) =>
        JobApplicationScoreUtil.getOriginalDomainScore(
            s.score as number,
            s.score_type,
        ),
    );

    const reasoningAverageScore = sum(scores) / scores.length;

    return JobApplicationScoreMapper.mapJobApplicationScore(
        JobApplicationScoreUtil.getFormattedJobApplicationScore(
            reasoningAverageScore,
            JobApplicationScoreType.REASONING_AVG,
        ),
        {
            cognitive_result: {
                domain_score: reasoningAverageScore,
                domain_alias: JobApplicationScoreType.REASONING_AVG,
                domain_percentile: null,
                ingredient_weightage: null,
            },
        },
        JobApplicationScoreType.REASONING_AVG,
        JobApplicationScoreService.getScoreDimension(reasoningAverageScore),
        jobApplicationId,
        updatedBy,
        recipeId,
    );
};

const getCognitiveScores = async (
    assessments: JobApplicationQuestionnaireDto[],
    jobApplicationId: string,
    updatedBy: number,
    recipeId?: string,
): Promise<JobApplicationScore[]> => {
    try {
        const cognitiveScores = assessments.flatMap((assessment) => {
            if (!QuestionnaireUtil.isCognitiveQuestionnaire(assessment)) {
                return [];
            }

            return processCognitiveScore(
                assessment,
                jobApplicationId,
                updatedBy,
                recipeId,
            );
        });

        if (cognitiveScores.length === 0) {
            return [];
        }

        const reasoningAverage = await processReasoningAverage(
            cognitiveScores,
            jobApplicationId,
            updatedBy,
            recipeId,
        );

        return [...cognitiveScores, reasoningAverage];
    } catch (err) {
        logger.error('Failed to process cognitive score', {
            data: {
                assessments,
                job_application_id: jobApplicationId,
                recipe_id: recipeId,
            },
            err,
        });
        throw err;
    }
};

export const CognitiveScoreService = {
    getCognitiveScores,
    processReasoningAverage,
    processCognitiveScore,
};
