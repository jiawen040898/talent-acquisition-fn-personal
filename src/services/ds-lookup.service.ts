import { AWSConfig } from '@pulsifi/configs';
import {
    DSLookupAPIErrorCode,
    DSLookupAPIErrorMessage,
    DSLookupAPIErrorType,
} from '@pulsifi/constants';
import { ErrorDetails } from '@pulsifi/fn';
import axios from 'axios';

export interface GetPairwiseSimilarity {
    score: number;
    pairwise_score: {
        [key: string]: {
            [key: string]: number;
        };
    };
}

const getPairwiseSimilarity = async (
    main: JSON | string[] | null,
    secondary: string[] | null,
): Promise<GetPairwiseSimilarity | undefined> => {
    const payload = {
        main,
        secondary,
    };

    const path = `${AWSConfig().alb.dns}/lookup/ds/pairwise-similarity/score`;

    try {
        const res = await axios.post<GetPairwiseSimilarity>(path, payload);

        return res.data;
    } catch (err) {
        throw new FailedToGetPairwiseSimilarityFromDSLookupApi({
            error_codes: [
                DSLookupAPIErrorCode.FAILED_TO_GET_PAIRWISE_SIMILARITY,
            ],
            err,
        });
    }
};

export class FailedToGetPairwiseSimilarityFromDSLookupApi extends Error {
    errorDetails: ErrorDetails;

    constructor(errorDetails: ErrorDetails) {
        super(DSLookupAPIErrorMessage.FAILED_TO_GET_PAIRWISE_SIMILARITY);
        this.name = DSLookupAPIErrorType.FAILED_TO_GET_PAIRWISE_SIMILARITY;
        this.errorDetails = errorDetails;
    }
}

export const DSLookupService = {
    getPairwiseSimilarity,
};
