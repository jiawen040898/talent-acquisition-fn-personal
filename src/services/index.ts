export * from './beta-distribution.service';
export * from './cognitive-score.service';
export * from './ds-lookup.service';
export * from './feature-toggle.service';
export * from './fit-score.service';
export * from './job-application.service';
export * from './job-application-score.service';
export * from './job-application-skill-extraction.service';
export * from './job-distribution-score.service';
export * from './job-user-access.service';
export * from './pairwise-score.service';
export * from './personality-score.service';
export * from './psychology.service';
export * from './publisher.service';
export * from './s3.service';
export * from './skill-extract-api.service';
