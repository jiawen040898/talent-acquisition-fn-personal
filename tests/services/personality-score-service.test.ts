import { DomainFramework, DomainOutcome } from '@pulsifi/fn';
import { PersonalityScoreService } from '@pulsifi/services';
import {
    personalityQuestionnaire,
    workInterestQuestionnaire,
    workValueQuestionnaire,
} from '@pulsifi/tests/fixtures/job-application-questionnaire/test-data';
import { mockRoleFitRecipe } from '@pulsifi/tests/fixtures/recipe/test-data';

const jobApplicationId = '00000000-0000-0000-0000-000000000001';

describe('PersonalityScoreService', () => {
    describe('getPersonalityScores', () => {
        it('should output partial JobApplicationScore', () => {
            // Act
            const result = PersonalityScoreService.getPersonalityScores(
                [
                    personalityQuestionnaire,
                    workInterestQuestionnaire,
                    workValueQuestionnaire,
                ],
                jobApplicationId,
                5,
                mockRoleFitRecipe.id,
                mockRoleFitRecipe,
            );

            // Assert
            expect(result).toMatchSnapshot();
        });

        it('should work correctly without recipe', () => {
            // Act
            const result = PersonalityScoreService.getPersonalityScores(
                [
                    personalityQuestionnaire,
                    workInterestQuestionnaire,
                    workValueQuestionnaire,
                ],
                jobApplicationId,
                5,
            );

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe('getWorkStyleScoreOutcome', () => {
        it('should return correct output', () => {
            // Act
            const result = PersonalityScoreService.getWorkStyleScoreOutcome(
                personalityQuestionnaire,
                jobApplicationId,
                5,
                mockRoleFitRecipe.id,
                mockRoleFitRecipe,
            );

            // Assert
            expect(result).toMatchSnapshot();
        });

        it('should return correct output without recipe', () => {
            // Act
            const result = PersonalityScoreService.getWorkStyleScoreOutcome(
                personalityQuestionnaire,
                jobApplicationId,
                5,
            );

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe('getWorkValueScoreOutcome', () => {
        it('should return correct output', () => {
            // Act
            const result = PersonalityScoreService.getWorkValueScoreOutcome(
                workValueQuestionnaire,
                jobApplicationId,
                5,
                mockRoleFitRecipe.id,
                mockRoleFitRecipe,
            );

            // Assert
            expect(result).toMatchSnapshot();
        });

        it('should return correct output without recipe', () => {
            // Act
            const result = PersonalityScoreService.getWorkValueScoreOutcome(
                workValueQuestionnaire,
                jobApplicationId,
                5,
            );

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe('getWorkInterestScoreOutcome', () => {
        it('should return correct output', () => {
            // Act
            const result = PersonalityScoreService.getWorkInterestScoreOutcome(
                workInterestQuestionnaire,
                jobApplicationId,
                5,
                mockRoleFitRecipe.id,
                mockRoleFitRecipe,
            );

            // Assert
            expect(result).toMatchSnapshot();
        });

        it('should return correct output without recipe', () => {
            // Act
            const result = PersonalityScoreService.getWorkInterestScoreOutcome(
                workInterestQuestionnaire,
                jobApplicationId,
                5,
            );

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe('transformDomainOutcome', () => {
        it('should return correct output', () => {
            // Arrange
            const domain: DomainOutcome = {
                domain_score: 0,
                domain_framework: DomainFramework.WORK_STYLE,
                domain_alias: 'verbal',
            };

            // Act
            const result =
                PersonalityScoreService.transformDomainOutcome(domain);

            // Assert
            expect(result).toEqual({
                domain_score: 0,
                domain_framework: DomainFramework.WORK_STYLE,
                domain_alias: 'verbal',
                domain_weightage: null,
                traits: [],
            });
        });

        it('should not replace weightage', () => {
            // Arrange
            const domain: DomainOutcome = {
                domain_score: 0,
                domain_framework: DomainFramework.WORK_STYLE,
                domain_alias: 'verbal',
                domain_weightage: 0.4,
            };

            // Act
            const result =
                PersonalityScoreService.transformDomainOutcome(domain);

            // Assert
            expect(result).toEqual({
                domain_score: 0,
                domain_framework: DomainFramework.WORK_STYLE,
                domain_alias: 'verbal',
                domain_weightage: 0.4,
                traits: [],
            });
        });
    });
});
