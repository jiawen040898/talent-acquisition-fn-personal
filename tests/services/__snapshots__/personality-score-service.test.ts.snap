// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalityScoreService getPersonalityScores should output partial JobApplicationScore 1`] = `
[
  {
    "created_by": 5,
    "job_application_id": "00000000-0000-0000-0000-000000000001",
    "score": 6.010443,
    "score_dimension": 1,
    "score_outcome": {
      "personality_result": [
        {
          "domain_alias": "achievement_effort",
          "domain_score": 0.8229166666666666,
          "domain_weightage": 0.066666667,
          "traits": [],
        },
        {
          "domain_alias": "persistence",
          "domain_score": 0.5625,
          "domain_weightage": 0.066666667,
          "traits": [],
        },
        {
          "domain_alias": "initiative",
          "domain_score": 0.7232142857142857,
          "domain_weightage": 0.066666667,
          "traits": [],
        },
        {
          "domain_alias": "leadership",
          "domain_score": 0.5416666666666666,
          "domain_weightage": 0.066666667,
          "traits": [],
        },
        {
          "domain_alias": "cooperation",
          "domain_score": 0.5,
          "domain_weightage": 0.033333333,
          "traits": [],
        },
        {
          "domain_alias": "concern_for_others",
          "domain_score": 0.6666666666666667,
          "domain_weightage": 0.066666667,
          "traits": [],
        },
        {
          "domain_alias": "social_orientation",
          "domain_score": 0.5104166666666666,
          "domain_weightage": 0.066666667,
          "traits": [],
        },
        {
          "domain_alias": "self_control",
          "domain_score": 0.5125,
          "domain_weightage": 0.033333333,
          "traits": [],
        },
        {
          "domain_alias": "stress_tolerance",
          "domain_score": 0.625,
          "domain_weightage": 0.066666667,
          "traits": [],
        },
        {
          "domain_alias": "adaptability_flexibility",
          "domain_score": 0.6796875,
          "domain_weightage": 0.1,
          "traits": [],
        },
        {
          "domain_alias": "dependability",
          "domain_score": 0.625,
          "domain_weightage": 0.066666667,
          "traits": [],
        },
        {
          "domain_alias": "attention_to_detail",
          "domain_score": 0.4375,
          "domain_weightage": 0.066666667,
          "traits": [],
        },
        {
          "domain_alias": "integrity",
          "domain_score": 0.5833333333333334,
          "domain_weightage": 0.1,
          "traits": [],
        },
        {
          "domain_alias": "independence",
          "domain_score": 0.6145833333333334,
          "domain_weightage": 0,
          "traits": [],
        },
        {
          "domain_alias": "innovation",
          "domain_score": 0.6625,
          "domain_weightage": 0.066666667,
          "traits": [],
        },
        {
          "domain_alias": "analytical_thinking",
          "domain_score": 0.4375,
          "domain_weightage": 0.066666667,
          "traits": [],
        },
      ],
    },
    "score_recipe_id": "106e0b1d-02c4-4321-9dac-9768f84f8bc1",
    "score_type": "work_style",
    "updated_by": 5,
  },
  {
    "created_by": 5,
    "job_application_id": "00000000-0000-0000-0000-000000000001",
    "score": 5.7143,
    "score_dimension": 1,
    "score_outcome": {
      "job_codes": [
        "I",
        "C",
        "E",
      ],
      "person_codes": [
        [
          "R",
          "I",
          "E",
        ],
        [
          "R",
          "E",
          "I",
        ],
        [
          "I",
          "R",
          "E",
        ],
        [
          "I",
          "E",
          "R",
        ],
        [
          "E",
          "R",
          "I",
        ],
        [
          "E",
          "I",
          "R",
        ],
      ],
      "personality_result": [
        {
          "domain_alias": "realistic",
          "domain_score": 0.71875,
        },
        {
          "domain_alias": "investigative",
          "domain_score": 0.71875,
        },
        {
          "domain_alias": "artistic",
          "domain_score": 0.46875,
        },
        {
          "domain_alias": "social",
          "domain_score": 0.625,
        },
        {
          "domain_alias": "enterprising",
          "domain_score": 0.71875,
        },
        {
          "domain_alias": "conventional",
          "domain_score": 0.53125,
        },
      ],
    },
    "score_recipe_id": "106e0b1d-02c4-4321-9dac-9768f84f8bc1",
    "score_type": "work_interest",
    "updated_by": 5,
  },
  {
    "created_by": 5,
    "job_application_id": "00000000-0000-0000-0000-000000000001",
    "score": 5.204422,
    "score_dimension": 1,
    "score_outcome": {
      "personality_result": [
        {
          "domain_alias": "achievement",
          "domain_score": 0.6071428571428571,
          "domain_weightage": 0.285714286,
          "traits": [],
        },
        {
          "domain_alias": "working_conditions",
          "domain_score": 0.49523809523809526,
          "domain_weightage": 0,
          "traits": [],
        },
        {
          "domain_alias": "recognition",
          "domain_score": 0.4083333333333334,
          "domain_weightage": 0,
          "traits": [],
        },
        {
          "domain_alias": "relationships",
          "domain_score": 0.49833333333333335,
          "domain_weightage": 0.285714286,
          "traits": [],
        },
        {
          "domain_alias": "support",
          "domain_score": 0.49166666666666664,
          "domain_weightage": 0.285714286,
          "traits": [],
        },
        {
          "domain_alias": "independence",
          "domain_score": 0.4488095238095238,
          "domain_weightage": 0.142857143,
          "traits": [],
        },
      ],
    },
    "score_recipe_id": "106e0b1d-02c4-4321-9dac-9768f84f8bc1",
    "score_type": "work_value",
    "updated_by": 5,
  },
]
`;

exports[`PersonalityScoreService getPersonalityScores should work correctly without recipe 1`] = `
[
  {
    "created_by": 5,
    "job_application_id": "00000000-0000-0000-0000-000000000001",
    "score": null,
    "score_dimension": 0,
    "score_outcome": {
      "personality_result": [
        {
          "domain_alias": "achievement_effort",
          "domain_score": 0.8229166666666666,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "persistence",
          "domain_score": 0.5625,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "initiative",
          "domain_score": 0.7232142857142857,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "leadership",
          "domain_score": 0.5416666666666666,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "cooperation",
          "domain_score": 0.5,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "concern_for_others",
          "domain_score": 0.6666666666666667,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "social_orientation",
          "domain_score": 0.5104166666666666,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "self_control",
          "domain_score": 0.5125,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "stress_tolerance",
          "domain_score": 0.625,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "adaptability_flexibility",
          "domain_score": 0.6796875,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "dependability",
          "domain_score": 0.625,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "attention_to_detail",
          "domain_score": 0.4375,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "integrity",
          "domain_score": 0.5833333333333334,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "independence",
          "domain_score": 0.6145833333333334,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "innovation",
          "domain_score": 0.6625,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "analytical_thinking",
          "domain_score": 0.4375,
          "domain_weightage": null,
          "traits": [],
        },
      ],
    },
    "score_recipe_id": undefined,
    "score_type": "work_style",
    "updated_by": 5,
  },
  {
    "created_by": 5,
    "job_application_id": "00000000-0000-0000-0000-000000000001",
    "score": null,
    "score_dimension": 0,
    "score_outcome": {
      "personality_result": [
        {
          "domain_alias": "realistic",
          "domain_score": 0.71875,
        },
        {
          "domain_alias": "investigative",
          "domain_score": 0.71875,
        },
        {
          "domain_alias": "artistic",
          "domain_score": 0.46875,
        },
        {
          "domain_alias": "social",
          "domain_score": 0.625,
        },
        {
          "domain_alias": "enterprising",
          "domain_score": 0.71875,
        },
        {
          "domain_alias": "conventional",
          "domain_score": 0.53125,
        },
      ],
    },
    "score_recipe_id": undefined,
    "score_type": "work_interest",
    "updated_by": 5,
  },
  {
    "created_by": 5,
    "job_application_id": "00000000-0000-0000-0000-000000000001",
    "score": null,
    "score_dimension": 0,
    "score_outcome": {
      "personality_result": [
        {
          "domain_alias": "achievement",
          "domain_score": 0.6071428571428571,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "independence",
          "domain_score": 0.4488095238095238,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "recognition",
          "domain_score": 0.4083333333333334,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "relationships",
          "domain_score": 0.49833333333333335,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "support",
          "domain_score": 0.49166666666666664,
          "domain_weightage": null,
          "traits": [],
        },
        {
          "domain_alias": "working_conditions",
          "domain_score": 0.49523809523809526,
          "domain_weightage": null,
          "traits": [],
        },
      ],
    },
    "score_recipe_id": undefined,
    "score_type": "work_value",
    "updated_by": 5,
  },
]
`;

exports[`PersonalityScoreService getWorkInterestScoreOutcome should return correct output 1`] = `
{
  "created_by": 5,
  "job_application_id": "00000000-0000-0000-0000-000000000001",
  "score": 5.7143,
  "score_dimension": 1,
  "score_outcome": {
    "job_codes": [
      "I",
      "C",
      "E",
    ],
    "person_codes": [
      [
        "R",
        "I",
        "E",
      ],
      [
        "R",
        "E",
        "I",
      ],
      [
        "I",
        "R",
        "E",
      ],
      [
        "I",
        "E",
        "R",
      ],
      [
        "E",
        "R",
        "I",
      ],
      [
        "E",
        "I",
        "R",
      ],
    ],
    "personality_result": [
      {
        "domain_alias": "realistic",
        "domain_score": 0.71875,
      },
      {
        "domain_alias": "investigative",
        "domain_score": 0.71875,
      },
      {
        "domain_alias": "artistic",
        "domain_score": 0.46875,
      },
      {
        "domain_alias": "social",
        "domain_score": 0.625,
      },
      {
        "domain_alias": "enterprising",
        "domain_score": 0.71875,
      },
      {
        "domain_alias": "conventional",
        "domain_score": 0.53125,
      },
    ],
  },
  "score_recipe_id": "106e0b1d-02c4-4321-9dac-9768f84f8bc1",
  "score_type": "work_interest",
  "updated_by": 5,
}
`;

exports[`PersonalityScoreService getWorkInterestScoreOutcome should return correct output without recipe 1`] = `
{
  "created_by": 5,
  "job_application_id": "00000000-0000-0000-0000-000000000001",
  "score": null,
  "score_dimension": 0,
  "score_outcome": {
    "personality_result": [
      {
        "domain_alias": "realistic",
        "domain_score": 0.71875,
      },
      {
        "domain_alias": "investigative",
        "domain_score": 0.71875,
      },
      {
        "domain_alias": "artistic",
        "domain_score": 0.46875,
      },
      {
        "domain_alias": "social",
        "domain_score": 0.625,
      },
      {
        "domain_alias": "enterprising",
        "domain_score": 0.71875,
      },
      {
        "domain_alias": "conventional",
        "domain_score": 0.53125,
      },
    ],
  },
  "score_recipe_id": undefined,
  "score_type": "work_interest",
  "updated_by": 5,
}
`;

exports[`PersonalityScoreService getWorkStyleScoreOutcome should return correct output 1`] = `
{
  "created_by": 5,
  "job_application_id": "00000000-0000-0000-0000-000000000001",
  "score": 6.010443,
  "score_dimension": 1,
  "score_outcome": {
    "personality_result": [
      {
        "domain_alias": "achievement_effort",
        "domain_score": 0.8229166666666666,
        "domain_weightage": 0.066666667,
        "traits": [],
      },
      {
        "domain_alias": "persistence",
        "domain_score": 0.5625,
        "domain_weightage": 0.066666667,
        "traits": [],
      },
      {
        "domain_alias": "initiative",
        "domain_score": 0.7232142857142857,
        "domain_weightage": 0.066666667,
        "traits": [],
      },
      {
        "domain_alias": "leadership",
        "domain_score": 0.5416666666666666,
        "domain_weightage": 0.066666667,
        "traits": [],
      },
      {
        "domain_alias": "cooperation",
        "domain_score": 0.5,
        "domain_weightage": 0.033333333,
        "traits": [],
      },
      {
        "domain_alias": "concern_for_others",
        "domain_score": 0.6666666666666667,
        "domain_weightage": 0.066666667,
        "traits": [],
      },
      {
        "domain_alias": "social_orientation",
        "domain_score": 0.5104166666666666,
        "domain_weightage": 0.066666667,
        "traits": [],
      },
      {
        "domain_alias": "self_control",
        "domain_score": 0.5125,
        "domain_weightage": 0.033333333,
        "traits": [],
      },
      {
        "domain_alias": "stress_tolerance",
        "domain_score": 0.625,
        "domain_weightage": 0.066666667,
        "traits": [],
      },
      {
        "domain_alias": "adaptability_flexibility",
        "domain_score": 0.6796875,
        "domain_weightage": 0.1,
        "traits": [],
      },
      {
        "domain_alias": "dependability",
        "domain_score": 0.625,
        "domain_weightage": 0.066666667,
        "traits": [],
      },
      {
        "domain_alias": "attention_to_detail",
        "domain_score": 0.4375,
        "domain_weightage": 0.066666667,
        "traits": [],
      },
      {
        "domain_alias": "integrity",
        "domain_score": 0.5833333333333334,
        "domain_weightage": 0.1,
        "traits": [],
      },
      {
        "domain_alias": "independence",
        "domain_score": 0.6145833333333334,
        "domain_weightage": 0,
        "traits": [],
      },
      {
        "domain_alias": "innovation",
        "domain_score": 0.6625,
        "domain_weightage": 0.066666667,
        "traits": [],
      },
      {
        "domain_alias": "analytical_thinking",
        "domain_score": 0.4375,
        "domain_weightage": 0.066666667,
        "traits": [],
      },
    ],
  },
  "score_recipe_id": "106e0b1d-02c4-4321-9dac-9768f84f8bc1",
  "score_type": "work_style",
  "updated_by": 5,
}
`;

exports[`PersonalityScoreService getWorkStyleScoreOutcome should return correct output without recipe 1`] = `
{
  "created_by": 5,
  "job_application_id": "00000000-0000-0000-0000-000000000001",
  "score": null,
  "score_dimension": 0,
  "score_outcome": {
    "personality_result": [
      {
        "domain_alias": "achievement_effort",
        "domain_score": 0.8229166666666666,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "persistence",
        "domain_score": 0.5625,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "initiative",
        "domain_score": 0.7232142857142857,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "leadership",
        "domain_score": 0.5416666666666666,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "cooperation",
        "domain_score": 0.5,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "concern_for_others",
        "domain_score": 0.6666666666666667,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "social_orientation",
        "domain_score": 0.5104166666666666,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "self_control",
        "domain_score": 0.5125,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "stress_tolerance",
        "domain_score": 0.625,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "adaptability_flexibility",
        "domain_score": 0.6796875,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "dependability",
        "domain_score": 0.625,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "attention_to_detail",
        "domain_score": 0.4375,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "integrity",
        "domain_score": 0.5833333333333334,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "independence",
        "domain_score": 0.6145833333333334,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "innovation",
        "domain_score": 0.6625,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "analytical_thinking",
        "domain_score": 0.4375,
        "domain_weightage": null,
        "traits": [],
      },
    ],
  },
  "score_recipe_id": undefined,
  "score_type": "work_style",
  "updated_by": 5,
}
`;

exports[`PersonalityScoreService getWorkValueScoreOutcome should return correct output 1`] = `
{
  "created_by": 5,
  "job_application_id": "00000000-0000-0000-0000-000000000001",
  "score": 5.204422,
  "score_dimension": 1,
  "score_outcome": {
    "personality_result": [
      {
        "domain_alias": "achievement",
        "domain_score": 0.6071428571428571,
        "domain_weightage": 0.285714286,
        "traits": [],
      },
      {
        "domain_alias": "working_conditions",
        "domain_score": 0.49523809523809526,
        "domain_weightage": 0,
        "traits": [],
      },
      {
        "domain_alias": "recognition",
        "domain_score": 0.4083333333333334,
        "domain_weightage": 0,
        "traits": [],
      },
      {
        "domain_alias": "relationships",
        "domain_score": 0.49833333333333335,
        "domain_weightage": 0.285714286,
        "traits": [],
      },
      {
        "domain_alias": "support",
        "domain_score": 0.49166666666666664,
        "domain_weightage": 0.285714286,
        "traits": [],
      },
      {
        "domain_alias": "independence",
        "domain_score": 0.4488095238095238,
        "domain_weightage": 0.142857143,
        "traits": [],
      },
    ],
  },
  "score_recipe_id": "106e0b1d-02c4-4321-9dac-9768f84f8bc1",
  "score_type": "work_value",
  "updated_by": 5,
}
`;

exports[`PersonalityScoreService getWorkValueScoreOutcome should return correct output without recipe 1`] = `
{
  "created_by": 5,
  "job_application_id": "00000000-0000-0000-0000-000000000001",
  "score": null,
  "score_dimension": 0,
  "score_outcome": {
    "personality_result": [
      {
        "domain_alias": "achievement",
        "domain_score": 0.6071428571428571,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "independence",
        "domain_score": 0.4488095238095238,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "recognition",
        "domain_score": 0.4083333333333334,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "relationships",
        "domain_score": 0.49833333333333335,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "support",
        "domain_score": 0.49166666666666664,
        "domain_weightage": null,
        "traits": [],
      },
      {
        "domain_alias": "working_conditions",
        "domain_score": 0.49523809523809526,
        "domain_weightage": null,
        "traits": [],
      },
    ],
  },
  "score_recipe_id": undefined,
  "score_type": "work_value",
  "updated_by": 5,
}
`;
